src/main/scala/org/camunda/feel/impl/builtin/ConversionBuiltinFunctions.scala

```diff
@@ -101,6 +65,10 @@ object ConversionBuiltinFunctions {
         ValLocalDateTime(dateTime.toLocalDate().atTime(time))
       case List(ValDateTime(dateTime), ValTime(time)) =>
         ValDateTime(time.withDate(dateTime.toLocalDate()))
+      case List(ValLocalDateTime(date), ValString(timezone)) =>
+        ValDateTime(date.atZone(ZoneId.systemDefault()).withZoneSameInstant(ZoneId.of(timezone)))
```

Comment:
:x: We should not transform the given date-time to the system date-time. Instead, we can just take the given timezone.

Otherwise, the function result could be confusing depending on the system's timezone.

```suggestion
      case List(ValLocalDateTime(date), ValString(timezone)) =>
        ValDateTime(date.atZone(ZoneId.of(timezone)))
```